src/coreclr/vm/arm/stubs.cpp

```diff
@@ -1992,19 +1993,22 @@ void UMEntryThunkCode::Encode(BYTE* pTargetCode, void* pvSecretParam)
     m_pTargetCode = (TADDR)pTargetCode;
     m_pvSecretParam = (TADDR)pvSecretParam;
 
-    FlushInstructionCache(GetCurrentProcess(),&m_code,sizeof(m_code));
+    FlushInstructionCache(GetCurrentProcess(),&pEntryThunkCodeRX->m_code,sizeof(m_code));
 }
 
 #ifndef DACCESS_COMPILE
 
 void UMEntryThunkCode::Poison()
 {
-    m_pTargetCode = (TADDR)UMEntryThunk::ReportViolation;
+    ExecutableWriterHolder<UMEntryThunkCode> thunkWriterHolder(this, sizeof(UMEntryThunkCode));
+    UMEntryThunkCode *pThisRW = thunkWriterHolder.GetRW();
+
+    pThisRW->m_pTargetCode = (TADDR)UMEntryThunk::ReportViolation;
 
     // ldr r0, [pc + 8]
-    m_code[0] = 0x4802;
+    pThisRW->m_code[0] = 0x4802;
     // nop
-    m_code[1] = 0xbf00;
+    pThisRW->m_code[1] = 0xbf00;
```

Comment:
not part of your change, but aren't there constants defined for standard opcodes?

Reply:
I am not aware of such a thing in our codebase.